transport-classes-io_uring/src/main/java/io/netty/channel/uring/AbstractIoUringServerChannel.java

```diff
@@ -35,24 +35,53 @@
+        private final long acceptedAddressMemoryAddress;
+        private final long acceptedAddressLengthMemoryAddress;
+
+        AcceptedAddressMemory() {
+            acceptedAddressMemory = Buffer.allocateDirectWithNativeOrder(Native.SIZEOF_SOCKADDR_STORAGE);
+            acceptedAddressMemoryAddress = Buffer.memoryAddress(acceptedAddressMemory);
+            acceptedAddressLengthMemory = Buffer.allocateDirectWithNativeOrder(Long.BYTES);
+            // Needs to be initialized to the size of acceptedAddressMemory.
+            // See https://man7.org/linux/man-pages/man2/accept.2.html
+            acceptedAddressLengthMemory.putLong(0, Native.SIZEOF_SOCKADDR_STORAGE);
+            acceptedAddressLengthMemoryAddress = Buffer.memoryAddress(acceptedAddressLengthMemory);
+        }
 
+        void free() {
+            Buffer.free(acceptedAddressMemory);
+            Buffer.free(acceptedAddressLengthMemory);
+        }
+    }
+    private final AcceptedAddressMemory acceptedAddressMemory;
     private long acceptId;
 
     protected AbstractIoUringServerChannel(LinuxSocket socket, boolean active) {
         super(null, socket, active);
 
-        acceptedAddressMemory = Buffer.allocateDirectWithNativeOrder(Native.SIZEOF_SOCKADDR_STORAGE);
-        acceptedAddressMemoryAddress = Buffer.memoryAddress(acceptedAddressMemory);
-        acceptedAddressLengthMemory = Buffer.allocateDirectWithNativeOrder(Long.BYTES);
-        // Needs to be initialized to the size of acceptedAddressMemory.
-        // See https://man7.org/linux/man-pages/man2/accept.2.html
-        acceptedAddressLengthMemory.putLong(0, Native.SIZEOF_SOCKADDR_STORAGE);
-        acceptedAddressLengthMemoryAddress = Buffer.memoryAddress(acceptedAddressLengthMemory);
+        // We can only depend on the accepted address if multi-shot is not used.
+        // From the manpage:
+        //       The multishot variants allow an application to issue a single
+        //       accept request, which will repeatedly trigger a CQE when a
+        //       connection request comes in. Like other multishot type requests,
+        //       the application should look at the CQE flags and see if
+        //       IORING_CQE_F_MORE is set on completion as an indication of whether
+        //       or not the accept request will generate further CQEs. Note that
+        //       for the multishot variants, setting addr and addrlen may not make
+        //       a lot of sense, as the same value would be used for every accepted
+        //       connection. This means that the data written to addr may be
+        //       overwritten by a new connection before the application has had
+        //       time to process a past connection. If the application knows that a
+        //       new connection cannot come in before a previous one has been
+        //       processed, it may be used as expected.
+        if (IoUring.isIOUringAcceptNoWaitSupported()) {
```

Comment:
why `isIOUringAcceptNoWaitSupported`. I think it should be `isIOUringAcceptMultishotSupported`

Reply:
🤦 ... fixed